meson.build

```diff
@@ -0,0 +1,1449 @@
+        'on', build_machine.system(), 'in', meson.project_source_root(), '⭐')
+compiler = meson.get_compiler('cpp')
+message('Using Meson', meson.version(), ', ', compiler.get_id(), 'compiler and', compiler.get_linker_id(), 'linker')
+
+#
+# We need two versions of the main executable name because they are different on Windows:
+#    - We need a "target" name _without_ the '.exe' suffix for the executable() command because, on Windows, Meson will
+#      always add its own '.exe' suffix and we don't want to end up with '.exe.exe' as the suffix!
+#    - Everywhere else, we want the actual name of the file, including the '.exe' suffix.
+#
+# The default (on Windows and Linux) is to use Unix-style executable names, ie all lower case
+#
+mainExecutableTargetName = projectName
+if host_machine.system() == 'darwin'
+   # On Mac we don't need a suffix but have always capitalised the executable name because "Don't question the APPLE".
+   mainExecutableTargetName = capitalisedProjectName
+endif
+if host_machine.system() == 'windows'
+   mainExecutableFileName = mainExecutableTargetName + '.exe'
+else
+   mainExecutableFileName = mainExecutableTargetName
+endif
+
+testRunnerTargetName = mainExecutableTargetName + '_tests'
+
+#=======================================================================================================================
+#==================================================== Meson modules ====================================================
+#=======================================================================================================================
+# Import the Qt tools.  See https://mesonbuild.com/Qt5-module.html
+qt = import('qt5')
+# File System module - see https://mesonbuild.com/Fs-module.html
+fs = import('fs')
+
+#=======================================================================================================================
+#==================================================== Build options ====================================================
+#=======================================================================================================================
+prefix = get_option('prefix')
+
+#=======================================================================================================================
+#============================================== Frameworks and Libraries ===============================================
+#=======================================================================================================================
+#
+# It would be neat within the build system to automate the _installation_ and upgrade of libraries and frameworks on
+# which we depend.  However, I have yet to find a universal pain-free solution.
+#
+# Meson has its own dependency management system, Wrap, but the list of pre-provided projects at
+# https://mesonbuild.com/Wrapdb-projects.html does not include Qt, Boost, Xerces, Xalan or Valijson.
```

Comment:
Qt at least would likely be far, far too painful to reimplement.

Anything else is probably feasible, and you can submit a request or a PR to add it to the WrapDB. We'll be happy to accept it. :)

Reply:
One for the future! :)